pkg/template/nstemplatetier_test.go

```diff
@@ -0,0 +1,169 @@
+package template_test
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	fedcommon "sigs.k8s.io/kubefed/pkg/apis/core/common"
+	fedv1b1 "sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+)
+
+func TestGetNSTemplateTier(t *testing.T) {
+
+	// given
+	// Setup Scheme for all resources (required before adding objects in the fake client)
+	err := apis.AddToScheme(scheme.Scheme)
+	require.NoError(t, err)
+	logf.SetLogger(zap.Logger())
+	basicTier := &toolchainv1alpha1.NSTemplateTier{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
+			Kind:       "NSTemplateTier",
+		},
```

Comment:
I don't really think that we need to set the `APIVersion` and `Kind` as you are using the `NSTemplateTier` object directly, so it should have been set automatically.

Reply:
not sure if those values will be set automatically, but I can remove them from my test, indeed